packages/node_modules/@node-red/nodes/core/function/10-function.html

```diff
@@ -70,12 +129,54 @@
                 }
             });
 
+            this.initEditor = RED.editor.createEditor({
+                id: 'node-input-init-editor',
+                mode: 'ace/mode/nrjavascript',
+                value: $("#node-input-initialize").val(),
+                globals: {
+                    msg:true,
+                    context:true,
+                    RED: true,
+                    util: true,
+                    flow: true,
+                    global: true,
+                    console: true,
+                    Buffer: true,
+                    setTimeout: true,
+                    clearTimeout: true,
+                    setInterval: true,
+                    clearInterval: true
+                }
+            });
+
+            this.finalizeEditor = RED.editor.createEditor({
+                id: 'node-input-finalize-editor',
+                mode: 'ace/mode/nrjavascript',
+                value: $("#node-input-finalize").val(),
+                globals: {
+                    msg:true,
+                    context:true,
+                    RED: true,
+                    util: true,
+                    flow: true,
+                    global: true,
+                    console: true,
+                    Buffer: true,
+                    setTimeout: true,
+                    clearTimeout: true,
+                    setInterval: true,
+                    clearInterval: true
+                }
+            });
+
             RED.library.create({
                 url:"functions", // where to get the data from
                 type:"function", // the type of object the library is for
                 editor:this.editor, // the field name the main text body goes to
+                initEditor: this.initEditor, // editor for initializer
```

Comment:
I don't think we should be adding `initEditor` and `finalizeEditor` as part of API options to `RED.library.create()`. The goal is to keep the APIs clean from any node-specific requirements.

As an alternative, can I suggest the `fields` property be updated to support an object rather than just a name. This object can then provide a getter and setter for the field - which the library code can use rather than assume it should read `node-input-XYZ`.

For example, we would then have:

```
fields: [
   'name', 
   'outputs',
   {
      id: 'initalize',
      get: function() {
         return this.initEditor.getValue();
      },
      set: function(v) {
         this.initEditor.setValue(v, -1);
      }
   },
   ...
]
```

This will also give us the ability to reuse the library for other things that aren't necessarily node's with their key/value pairs.... but that is for a different day.

We then need to decide where the logic around multi-line content comes in. Options are:
 - make it part of the `get`/`set` function I propose above. Downside: a lot of possible repetition.
 - add a flag to the `fields` object to tell the library it is a multi-line field that should be handled properly
 - just have the library code test for `\n` in *all* fields and do the encoding if needed. Downside: encoding `\n` is easy - can we reliably detect when it needs decoding going the other way. The risk is we get a false-positive because a field contains `\\\\n` legitimately.

What do you think? On balance, I think adding a `multiline: true` flag to the fields object is the safest approach.

**Update:** Having written all of that, I do not think the multi-line encoding should be happening in the editor at all. I think that should be done in the storage implementation. If a storage implementation is able to support multi-line fields (for example YAML), then we should allow that.

Reply:
Considering that the output format will be changed to JSON or YAML in the future, I think it would be better to handle the format conversion on the library side.
On the other hand, regarding merger of three editors, I think it is necessary to change the `fields` property into an object array as you suggested.